core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -222,8 +222,8 @@ public static LoadTableResponse createTable(
     throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
   }
 
-  public static void dropTable(Catalog catalog, TableIdentifier ident) {
-    boolean dropped = catalog.dropTable(ident);
+  public static void dropTable(Catalog catalog, TableIdentifier ident, boolean purge) {
```

Comment:
We need to maintain the original signature as well for semver and add deprecation for 1.2.

Reply:
done